mkosi/util.py

```diff
@@ -250,8 +251,7 @@ def name(cls) -> str:
 
     @classmethod
     def home(cls) -> Path:
-        home = os.getenv("SUDO_HOME") or os.getenv("HOME") or pwd.getpwuid(cls.uid()).pw_dir
-        return Path(home)
+        return Path(f"~{cls.name()}").expanduser()
```

Comment:
When sudo/pkexec are not used, I feel like `Path.home()` (which is just `Path("~").expanduser()`) should be used (taking `$HOME` into account and not going through `pwd.getpwnam()`).

Maybe `name()` and `home()` could share something like:

```python
    @classmethod
    def _name_from_env(cls) -> str:
        if (uid := cls._uid_from_env()) is not None:
            return os.getenv("SUDO_USER") or pwd.getpwuid(uid).pw_name
        return ""

    @classmethod
    def name(cls) -> str:
        return cls._name_from_env() or getpass.getuser()

    @classmethod
    def home(cls) -> Path:
        return Path(f"~{cls._name_from_env()}").expanduser()
```